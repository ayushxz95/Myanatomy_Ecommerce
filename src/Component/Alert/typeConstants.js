export const mockItem = {
      state:5,
      id:1,
      type:"Electronics",
      statetype:"Electronics",
      Alertmessage:"Cancel",
      color:"error",
  };