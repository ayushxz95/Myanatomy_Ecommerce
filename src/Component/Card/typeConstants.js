export const mockItem = {
    name: "Fan",
    id: 1,
    type: "Electronics",
    company: "Havells",
    key: 1,
  };