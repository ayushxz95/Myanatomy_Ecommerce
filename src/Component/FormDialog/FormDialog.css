.text-field{
    display: flex;
    flex-direction: column;
}   
.Dialog-container {
    background-color: WHITE;
}
.select-field{
  margin-top: 6px;
  font-size: 1rem;
  line-height: 1.4375em;
  border-radius: 4px;
  outline: none;
  padding: 10px;
  border-bottom: 1px solid;
  /* margin-top: 8px;
  color: green; */
}
/* select:focus, select:hover {
  outline-color: green;
} */
.eye-container {
  position: relative;
}
.eye-field{
   position: absolute;
   right: 6px;
   top: 18px;
}
.disabled-button {
  border: none;
  color: gray;
  cursor: not-allowed;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
    font-weight: 500;
    font-size: 0.875rem;
    line-height: 1.75;
    letter-spacing: 0.02857em;
    text-transform: uppercase;
    min-width: 64px;
    padding: 6px 8px;
    border-radius: 4px;
}
