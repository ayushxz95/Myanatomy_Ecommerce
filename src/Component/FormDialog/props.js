import { addItems, updateItem } from "../../Redux/Actions/ItemAction";

const mapStateToProps = (state) => {
  return {
     category: state.category.category,
}
}
const mapDispatchToProps = (dispatch) => {
    return {
      updateItem: (state, id, type) => dispatch(updateItem(state, id, type)),
      addItems: (state, type) => dispatch(addItems(state, type)),
    };
  };
  
export  {mapStateToProps, mapDispatchToProps}