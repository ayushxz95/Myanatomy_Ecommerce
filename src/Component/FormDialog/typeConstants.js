export const mockItem = {
  message: "Update Product",
  buttonmessage: "Update",
  type: "Electronics",
  company: "Bajaj",
  name: "Fan",
  id: 3,
};
