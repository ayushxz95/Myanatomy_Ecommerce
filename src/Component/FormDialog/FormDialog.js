import * as React from "react";
import { Component } from "react";
import Button from "@mui/material/Button";
import TextField from "@mui/material/TextField";
import Dialog from "@mui/material/Dialog";
import DialogActions from "@mui/material/DialogActions";
import DialogContent from "@mui/material/DialogContent";
import { BsPencilFill } from "react-icons/bs";
import DialogTitle from "@mui/material/DialogTitle";
import { FiPlusSquare } from "react-icons/fi";
import { PropTypes } from "prop-types";
import { AiFillEye } from "react-icons/ai";
import Alert from "../Alert/index";
import "./FormDialog.css";

class FormDialog extends Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false,
      name: this.props.name,
      company: this.props.company,
      type: this.props.type,
      image: this.props.image,
    };
  }

  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  };

  handleClick = () => {
    this.setState((prev) => ({ open: !prev.open }));
  };

  onAddSubmit = () => {
    this.setState({
      name: "",
      company: "",
      type: "",
      image: "",
    });
  };

  onSubmit = () => {
    this.setState({
      name: this.props.name,
      company: this.props.company,
      type: this.props.type,
      image: this.props.image,
    });
  };

  render() {
    console.log(this.props.category);
    const { company, name, type, image } = this.state;
    const { buttonmessage } = this.props;
    const isCompanyNameTouched = company !== this.props.company ? true : false;
    const isNameTouched = name !== this.props.name ? true : false;
    const isTypeTouched = type !== this.props.type ? true : false;
    const isImageTouched = image !== this.props.image ? true : false;
    const isCompanyNameEmpty = company === "" ? true : false;
    const isNameEmpty = name === "" ? true : false;
    const isTypeEmpty = type === "" ? true : false;
    const isImageEmpty = image === "" ? true : false;
    const isNameValid = isNameTouched && !isNameEmpty ? true : false;
    const isCompanyNameValid =
      isCompanyNameTouched && !isCompanyNameEmpty ? true : false;
    const isTypeValid = isTypeTouched && !isTypeEmpty ? true : false;
    const isImageValid = isImageTouched && !isImageEmpty ? true : false;
    let disabled = true;
    if (isCompanyNameValid) {
      if (!isNameEmpty && !isTypeEmpty && !isImageEmpty) {
        disabled = false;
      }
    }
    if (isNameValid) {
      if (!isCompanyNameEmpty && !isTypeEmpty && !isImageEmpty) {
        disabled = false;
      }
    }
    if (isTypeValid) {
      if (!isCompanyNameEmpty && !isNameEmpty && !isImageEmpty) {
        disabled = false;
      }
    }
    if (isImageValid) {
      if (!isCompanyNameEmpty && !isTypeEmpty && !isNameEmpty) {
        disabled = false;
      }
    }
    if (type === "Select") {
      disabled = true;
    }
    if (company.length < 4) {
      disabled = true;
    }
    if(isImageEmpty === true){
      disabled = true;
    }
    let cancelbtn = false;
    if (isCompanyNameTouched || isTypeTouched || isNameTouched) {
      cancelbtn = true;
    }
    return (
      <div>
        <Button
          variant="text"
          onClick={this.handleClick}
          title={this.props.buttonmessage}
        >
          {this.props.buttonmessage === "Update" ? (
            <BsPencilFill size="18px" color="green" />
          ) : (
            <FiPlusSquare size="30px" color="black" />
          )}
        </Button>
        <Dialog open={this.state.open} onClose={this.handleClick}>
          <form className="Dialog-container">
            <DialogTitle color="#ef257a">{this.props.message}</DialogTitle>
            <DialogContent>
              <div className="text-field">
                {company.length > 3 ? (
                  <TextField
                    className="textfield"
                    autoFocus
                    required
                    margin="dense"
                    id="name"
                    label="Company"
                    value={company}
                    inputProps={{ minLength: 4 }}
                    onChange={this.onChange}
                    name="company"
                    type="text"
                    size="small"
                    variant="outlined"
                  />
                ) : (
                  <TextField
                    className="textfield"
                    autoFocus
                    required
                    margin="dense"
                    id="name"
                    label="Company"
                    value={company}
                    inputProps={{ minLength: 4 }}
                    onChange={this.onChange}
                    name="company"
                    type="text"
                    color="error"
                    size="small"
                    variant="outlined"
                  />
                )}
                {name.length > 3 ? (
                  <TextField
                    className="textfield"
                    required
                    id="name"
                    label="Name"
                    inputProps={{ minLength: 4 }}
                    value={name}
                    onChange={this.onChange}
                    name="name"
                    type="text"
                    size="small"
                    variant="outlined"
                    margin="dense"
                  />
                ) : (
                  <TextField
                    className="textfield"
                    required
                    id="name"
                    label="Name"
                    color="error"
                    inputProps={{ minLength: 4 }}
                    value={name}
                    onChange={this.onChange}
                    name="name"
                    type="text"
                    size="small"
                    variant="outlined"
                    margin="dense"
                  />
                )}
                <div className="eye-container">
                  <TextField
                    className="textfield"
                    InputProps={{
                      style: {
                        paddingRight: "14px",
                      },
                    }}
                    required
                    id="image"
                    label="Image"
                    value={image}
                    onChange={this.onChange}
                    name="image"
                    type="text"
                    size="small"
                    variant="outlined"
                    margin="dense"
                  />
                  <a
                    href={image}
                    target="_blank"
                    className="eye-field"
                    title="Preview"
                  >
                    <AiFillEye color="black" size="20px" />
                  </a>
                </div>
                {/* {this.props.buttonmessage === "Add" ? (
                  <select
                    className="select-field"
                    value={this.props.category}
                    name="type"
                    onChange={this.onChange}
                  >
                    <option value="Select">Select</option>
                    <option value="Grocery">Grocery</option>
                    <option value="Cloths">Cloths</option>
                    <option value="Electronics">Electronics</option>
                  </select>
                ) : ( */}
                  <select
                    className="select-field"
                    value={type}
                    name="type"
                    onChange={this.onChange}
                  >
                    <option value="Select">Select</option>
                    <option value="Grocery">Grocery</option>
                    <option value="Cloths">Cloths</option>
                    <option value="Electronics">Electronics</option>
                  </select>
                {/* )} */}
              </div>
            </DialogContent>
            <DialogActions>
              {cancelbtn === true ? (
                <Alert
                  state={this.state}
                  id={this.props.id}
                  type={this.props.type}
                  statetype={this.state.type}
                  Alertmessage={"Cancel"}
                  handleClick={this.handleClick}
                  color={"error"}
                  submit={this.onSubmit}
                  AddSubmit={this.onAddSubmit}
                />
              ) : (
                <Button color="error" onClick={this.handleClick}>
                  Cancel
                </Button>
              )}
              {disabled === false ? (
                <Alert
                  state={this.state}
                  disabled={disabled}
                  id={this.props.id}
                  type={this.props.type}
                  statetype={this.state.type}
                  Alertmessage={buttonmessage}
                  handleClick={this.handleClick}
                  color={"success"}
                  submit={this.onSubmit}
                  AddSubmit={this.onAddSubmit}
                />
              ) : (
                <button
                  className="disabled-button"
                  title="Please fill/change the fields"
                >
                  {buttonmessage}
                </button>
              )}
            </DialogActions>
          </form>
        </Dialog>
      </div>
    );
  }
}

FormDialog.propTypes = {
  updateItem: PropTypes.func,
  addItems: PropTypes.func,
};


export default FormDialog;
