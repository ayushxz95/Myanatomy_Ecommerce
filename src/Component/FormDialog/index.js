import { connect } from "react-redux";
import { mapStateToProps, mapDispatchToProps } from "./props";
import FormDialog from "./FormDialog";

export default connect(mapStateToProps, mapDispatchToProps)(FormDialog);