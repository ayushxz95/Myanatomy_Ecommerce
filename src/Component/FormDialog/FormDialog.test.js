import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import FormDialog from "./FormDialog";
import {mockItem} from "./typeConstants"

configure({ adapter: new Adapter() });


describe("Testing FormDialog Component", () => {
  it("Matching FormDialog's snapshot", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it("check for name field", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    expect(component.state("name")).toEqual(mockItem.name);
  });
  it("check for company field", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    expect(component.state("company")).toEqual(mockItem.company);
  });
  it("check for type field", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    expect(component.state("type")).toEqual(mockItem.type);
  });
  it("check for name field after changing", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    component.setState({name: "Cooler"})
    expect(component.state("name")).toEqual("Cooler");
  });
  it("check for company field after changing", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    component.setState({company: "Havells"})
    expect(component.state("company")).toEqual("Havells");
  });
  it("check for type field after changing", () => {
    const component = shallow(
      <FormDialog
              message={mockItem.message}
              buttonmessage={mockItem.buttonmessage}
              type={mockItem.type}
              company={mockItem.company}
              name={mockItem.name}
              id={mockItem.id}
      />
    );
    component.setState({type: "Electronics"})
    expect(component.state("type")).toEqual("Electronics");
  });
});