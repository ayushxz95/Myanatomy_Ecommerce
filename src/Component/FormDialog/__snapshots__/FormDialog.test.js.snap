// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FormDialog Component Matching FormDialog's snapshot 1`] = `
<div>
  <ForwardRef(Button)
    onClick={[Function]}
    title="Update"
    variant="text"
  >
    <BsPencilFill
      color="green"
      size="18px"
    />
  </ForwardRef(Button)>
  <ForwardRef(Dialog)
    onClose={[Function]}
    open={false}
  >
    <form
      className="Dialog-container"
    >
      <ForwardRef(DialogTitle)
        color="#ef257a"
      >
        Update Product
      </ForwardRef(DialogTitle)>
      <ForwardRef(DialogContent)>
        <div
          className="text-field"
        >
          <ForwardRef(TextField)
            autoFocus={true}
            className="textfield"
            id="name"
            label="Company"
            margin="dense"
            name="company"
            onChange={[Function]}
            required={true}
            size="small"
            type="text"
            value="Bajaj"
            variant="outlined"
          />
          <ForwardRef(TextField)
            autoFocus={true}
            className="textfield"
            id="name"
            label="Name"
            margin="dense"
            name="name"
            onChange={[Function]}
            required={true}
            size="small"
            type="text"
            value="Fan"
            variant="outlined"
          />
          <div
            className="eye-container"
          >
            <ForwardRef(TextField)
              InputProps={
                Object {
                  "style": Object {
                    "paddingRight": "14px",
                  },
                }
              }
              autoFocus={true}
              className="textfield"
              id="image"
              label="Image"
              margin="dense"
              name="image"
              onChange={[Function]}
              required={true}
              size="small"
              type="text"
              variant="outlined"
            />
            <a
              className="eye-field"
              target="_blank"
              title="Preview"
            >
              <AiFillEye
                color="black"
                size="20px"
              />
            </a>
          </div>
          <select
            className="select-field"
            name="type"
            onChange={[Function]}
            value="Electronics"
          >
            <option
              value="Select"
            >
              Select
            </option>
            <option
              value="Grocery"
            >
              Grocery
            </option>
            <option
              value="Cloths"
            >
              Cloths
            </option>
            <option
              value="Electronics"
            >
              Electronics
            </option>
          </select>
        </div>
      </ForwardRef(DialogContent)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button)
          color="error"
          onClick={[Function]}
        >
          Cancel
        </ForwardRef(Button)>
        <button
          className="disabled-button"
          title="Please fill/change the fields"
        >
          Update
        </button>
      </ForwardRef(DialogActions)>
    </form>
  </ForwardRef(Dialog)>
</div>
`;
